CRT.c

```diff
@@ -685,11 +689,94 @@ void CRT_restorePrivileges() {
 
 #endif /* HAVE_SETUID_ENABLED */
 
+#ifndef NDEBUG
+
+static int stderrRedirectNewFd = -1;
+static int stderrRedirectBackupFd = -1;
+
+static int createStderrCacheFile(void) {
+#ifdef HAVE_MEMFD_CREATE
+   return memfd_create("htop.stderr-redirect", 0);
+#else
+   char tmpName[] = "htop.stderr-redirectXXXXXX";
+   int r = mkstemp(tmpName);
```

Comment:
For Linux kernels 3.11+ it may be of interest to use `open`(2) with `O_TMPFILE` to create the tempfile with proper flags so it won't even hit the disk at all.

Reply:
`memfd_create(2)` exists since 3.17, do we really want to add extra code for 3.11-3.17 ?
`O_TMPFILE` is a Linux specific extension and probably only developers are using the debug mode and probbaly only on more recent Linux versions.

Reply:
Having the file available in the file system may constitute attack surface. With `O_TMPFILE` we get a situation that is quite similar to what `memfd_create` offers.

Reply:
Added an `O_TMPFILE` alternative.